development.md

```diff
@@ -0,0 +1,74 @@
+## Development
+
+This doc explains how to set up a development environment so you can get started contributing to this project
+
+## Install Requirements
+
+1. [`go`](https://golang.org/doc/install): For building the project
+1. [`git`](https://help.github.com/articles/set-up-git/): For source control
+1. [`gcloud`](https://cloud.google.com/sdk/docs/install): For Interacting with the Google Cloud Platform
+
+
+## Setup
+
+1. First Go ahead and create new project in the Google Cloud Platform and enable billing for the project
+
+2. Once you have created the project enable the below apis 
+
+```
+gcloud services enable compute.googleapis.com
+gcloud services enable container.googleapis.com
+```
+
+3. Next go ahead and create the service account in the project and give necessary permission and create the service account key by following this [link](https://cloud.google.com/iam/docs/creating-managing-service-account-keys)
```

Comment:
Which are the permissions?

Reply:
I think we can open another PR to limit the permission, and then update this. BTW, I add a task in the issue: 
https://github.com/cloudpilot-ai/karpenter-provider-gcp/issues/16

![image](https://github.com/user-attachments/assets/9701573b-af40-472f-9ff2-45a9a7a4dc8b)

Reply:
Hey @froblesmartin  
Thanks for the suggestion , Since I am new community I don't understand the full scope of the projects so I left it as it is , But once I get the full understanding of the project may be I'll reduce it to the least privileges

Reply:
@jwcesign for now can I add to the docs to add `owner` permission and add a note saying that we are working on the fix for proper permission will that works ?

Reply:
Sure

Reply:
@jwcesign  this has been updated